[Chong Jun Xiang] iP

Chongjx opened the pull request.

Chongjx committed:
Level-0

Added greet with simple layout

Chongjx committed:
Modified greeting message

Changed Duke's name to Jay
Added date and time into greeting message

Chongjx committed:
Modified classes and functions to retrieve date and time

Added DateTimeManager Class, shift the previous getDate/Time/Day functions into this class.
Added Task, TaskManager and IOManager classes for future implementation

Chongjx committed:
Added a function to format the printing of reply

Chongjx committed:
Added Echo and Exit command

Echos commands entered by the user and exits when the user types bye

Chongjx committed:
Implmented task list and adding of task

Added enum for the different possible types of user commands, reformatted reply message

Chongjx committed:
Implemented Task class

Able to mark done for tasks, deleted TaskManager class since the Duke class serve as the overall manager

Chongjx committed:
Refactored code to follow coding standards

Chongjx committed:
Added Todo, Deadline, Event classes

Todo, Deadline and event classes inherit from Task class

Chongjx committed:
Implemented automated testing of Text UIs

TC001
Tested all the functionality of the Todo, Event, Deadline class

R-Ramana commented on src/main/java/DateTimeManager.java:
```diff
@@ -0,0 +1,18 @@
+import java.time.format.DateTimeFormatter;
+
+public class DateTimeManager {
+    private static LocalDateTime current = LocalDateTime.now();
```
I think perhaps currentTime or something to that effect might be better in terms of clarifying what the current variable refers to.

Chongjx replied:
Updated the naming, thanks!

R-Ramana commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,185 @@
+    }
+
+    private static void printWelcomeMessage() {
+        // Welcome message
```
Seems like your comments for the above methods were all written in the line above the method. Perhaps to be consistent, you could also have this comment in the line just above.

Chongjx replied:
Noted on this, thanks!

R-Ramana commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,185 @@
+
+    // List task
+    private static void ListTask() {
+        int index = 1;
```
I see the index was declared here, but only used in the else statement. Would it be better if we declared it over there instead?

Chongjx replied:
Noted this too, thanks!

R-Ramana reviewed (commented):
Great job! Followed the coding standards very well.

Chongjx committed:
Rename some of the variables for better readability, standardised some of the display and comment

DateTimeManager class: current -> now
Deadline class: output from 'by' -> 'By' (standardised cap with the output from Event class)
Event class: output from 'On' -> 'At' (standardised with the keyword that the user input)

Cao-Zeyu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,183 @@
+    private static List<Task> taskList = new ArrayList<>();
+
+    // Print message with formatting
+    private static void reply(String msg) {
```
It might be better to avoid abbreviation?

Chongjx replied:
Noted, thanks!

Cao-Zeyu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,183 @@
+    }
+
+    // Add task
+    private static void AddTask() {
```
You can consider following the camelCase format for methods where the first letter is lowercase.

Chongjx replied:
Thanks for spotting this! Changed all my functions naming!

Cao-Zeyu commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Override
+    public String toString() {
+        return "[" + (isDone ? Duke.tick : Duke.cross) + "] " + description;
```
To increase the code readability, you can replace the (... ? ... : ...) statement with an if-else statement. But line 33 still seems quite okay to me.

Chongjx replied:
Noted, thanks!

Cao-Zeyu reviewed (commented):
Good job! :)

Chongjx committed:
Modified the code to follow the coding standard

Renamed some of the functions to follow the camelCase format

Chongjx committed:
Implement error handling

Types of error handling:
1. Invalid user command
2. Missing task description
3. Missing task identifier for deadline and event
4. Missing date/time info for deadline and event
5. Invalid input when marking task done

Chongjx committed:
Remove unwanted package

Chongjx committed:
Merge branch 'branch-Level-5'

Chongjx committed:
Organized the classes into packages

Chongjx committed:
Repackage 2 classes

DukeException and Formatter into util package

Chongjx committed:
Restructured the code and created a TaskManager.java

Moved task-related functions from Duke.java into TaskManager.java

Chongjx committed:
Starting working on IOManager

Created sample txt file to store the data

Chongjx committed:
Implemented deletion of task

Added more comments for the functions in TaskManager.java, explaining the return type and parameters.
Improved readability by giving proper names to the variables
Updated the EXPECTED.TXT according to the new format

Chongjx committed:
Implemented saving and loading of txt file.

Edited Task, Todo, Event and Deadline to return single char String for taskType for easier reading from text file
CAPs all the constant in Formatter.java

Chongjx committed:
Merge branch 'branch-Level-6'

Chongjx committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/manager/TaskManager.java
#	src/main/java/duke/util/Formatter.java

Chongjx committed:
Updated gitignore to stop tracking manifest file

Chongjx committed:
Renamed some of the command as well as the variables used for formatting and indentation. Updated text-ui-testing to handle loading from file

leowweixiang commented on src/main/java/duke/manager/TaskManager.java:
```diff
@@ -0,0 +1,165 @@
+    // Common functionality for markTaskDone and deleteTask
+    // Param: boolean isMarkDone, true for marking task done; false for deleting task
+    // Param: String[] taskIndexString, contains the whole string of the original message. Index 1 is the task index
+    private void markDoneOrDeleteTask(boolean isMarkDone, String[] taskIndexString) {
```
Interesting, but will recommend splitting these into done() and delete(), since they're quite different operations. Even though you can recycle some exception code, you still have implemented done() and delete() wrappers anyway

Chongjx replied:
Hi, noted on this. Originally I split these into 2 functions, however the codes are the same, except for the message that is to be printed out and that is why I created this function to factor out the code. Will made the necessary changes.

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,140 @@
+        String command = splitMessage[0];
+
+        switch (command.toUpperCase()) {
+        case "LIST":
```
Being nitpicky, but I'll suggest placing these as static finals somewhere instead of being magic strings, esp. since "TODO" is used again in addTask

Chongjx replied:
Noted on this, thanks!

leowweixiang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,140 @@
+        default:
+            replyMessage = "";
+        }
+        reply(replyMessage);
```
Consider returning the replyMessage like with your other commands, instead of reply() here, so that you can consolidate all the reply() calls together

Chongjx replied:
Noted on this.

leowweixiang commented on src/main/java/duke/manager/TaskManager.java:
```diff
@@ -0,0 +1,165 @@
+            // Create new task object based on the command type
+            switch (command.toUpperCase()) {
+            case "DEADLINE":
+                taskInfo = validateTaskInfo(message, "/by");
```
One suggestion to avoid magic strings again, is to put the identifiers as a static final in the specific classes, e.g. validateTaskInfo(message, Deadline.IDENTIFIER)

Chongjx replied:
Thanks for the tip!

leowweixiang commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,26 @@
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        this.taskType = "D";
```
Since Task has taskType getter/setters there, I'll suggest using those. Also, since Task is now an abstract class, I'll suggest having the super() handle the taskType too

Chongjx replied:
Noted on this, tks!

leowweixiang reviewed (commented):
Overall,, I think it adheres to coding guidelines well. No deep nesting, Formatter for centralising the UI strings used, clear "happy path".

One general thing: I find the returnMessage building "return +=" inside a method hard to read at times, but I don't think it's a guideline violation, so IMO it should be fine.

Chongjx committed:
Cleaned up codes based on TA feedback

Replaced magic strings with static final Strings.
Split the function of marking task done and deleting task.
Updated text-ui-test to test delete function as well

Chongjx committed:
Rename some of the static final variables for better readability.

Standardised all static final variable declaration.
Updated to save to txt file whenever there is any changes.
Added user generated file to gitignore

Chongjx committed:
Changed all the comments to follow javadoc format

Chongjx committed:
Created an UI class to handle interactions with the user.

Chongjx committed:
Tidying up for code for moreOOP

Chongjx committed:
Added Parser class and finished tidying up of code to make it more OOP.

Updated text-ui-test input and expected text files.

Chongjx committed:
Implemented handling of date and time for events and tasks.

Allow multiple formats of date and time entered.
Added a new command "On" to print events and deadlines occuring on a specific date.
Updated IOManager to read and write LocalDateTime instead of string for events and deadlines.
Added new parser methods for parse LocalDateTime and LocalDate

Chongjx committed:
Edited comment

Chongjx committed:
Implemented a find functionality with the command "Find".

Chongjx committed:
Updated some header comment

Chongjx committed:
Merge pull request #1 from Chongjx/branch-Level-8

Branch level 8

Chongjx committed:
Merge branch 'master' of https://github.com/Chongjx/ip

Chongjx committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/duke/util/Parser.java

Chongjx committed:
Merge pull request #2 from Chongjx/branch-Level-9

Branch level 9

Chongjx committed:
Merge branch 'master' of https://github.com/Chongjx/ip

Chongjx committed:
Merge branch 'master' into A-JavaDoc

# Conflicts:
#	src/main/java/duke/managers/DateTimeManager.java

Chongjx committed:
Merge pull request #3 from Chongjx/A-JavaDoc

A java doc

Chongjx committed:
Merge branch 'master' of https://github.com/Chongjx/ip

Chongjx committed:
Updated header comments

Chongjx committed:
Set theme jekyll-theme-merlot

Chongjx committed:
Removed the UIManager handler in IOManager and TaskManager.

All printing is done in the handleUserInput function in Duke.
Updated the text-ui-test to test for "On" and "Find" function.
Updated javaDoc.
Working on README.md

Chongjx committed:
For README.md

Chongjx committed:
Updated README.md

Chongjx committed:
Updated README.md V1.01

Chongjx committed:
Updated README.md V1.02

Chongjx committed:
Updated README.md V1.03

Chongjx committed:
Updated README.md V1.04

Chongjx committed:
Updated some of the comments for clearer description

Chongjx committed:
Updated README.md V1.04

Chongjx committed:
Added error handling for corrupted data in text file
